snuba/datasets/storages/selectors/errors.py

```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+from snuba import state
+from snuba.datasets.storage import StorageAndMappers
+from snuba.datasets.storages.factory import get_storage, get_writable_storage
+from snuba.datasets.storages.selectors.selector import QueryStorageSelector
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.query.logical import Query
+from snuba.query.query_settings import QuerySettings
+
+
+class ErrorsQueryStorageSelector(QueryStorageSelector):
+    def __init__(self) -> None:
+        self.__errors_table = get_writable_storage(StorageKey.ERRORS)
+        self.__errors_ro_table = get_storage(StorageKey.ERRORS_RO)
```

Comment:
This pattern is duplicating knowledge that is already in the configuration and then using `get_storage_mapping_pair` to do this reverse lookup, it's not very intuitive. I think there are more egonomic ways to do this. 

For example, we can provide the guarantee that `storage_and_mappers_list` is always going to be in the order that it is defined in configuration, that way you can do something like:


```python3
   def select_storage(
        self,
        query: Query,
        query_settings: QuerySettings,
        storage_and_mappers_list: List[StorageAndMappers],
    ) -> StorageAndMappers:
        use_readonly_storage = (
            state.get_config("enable_events_readonly_table", False)
            and not query_settings.get_consistent()
        )
        errors_ro, errors = storage_and_mappers_list
        if use_readonly_storage:
            return errors_ro
        return errors
```

That way you don't need to be worrying about implementing a good reverse mapping function and the coherence of the configuration is preserved

Reply:
Right, this is duplicating knowledge that is already in configuration, but it's also the way we do things today (selectors define storages in its constructor). However, I'm also not a fan of the reverse lookup. 

Guaranteeing order is an interesting approach (e.g. sorting storages by some condition). Is what you're suggesting, read-only storages should always be defined first in the array, followed by writable storages? What if there are multiple readable storages? How should they be ordered?

To summarize the problem, it is the job of `select_storage()` to determine which storage (defined on entity) to use based on input parameters. If the selection choice is always binary (readable or writable), it could probably just check `isinstance()` to determine which storage in the array (len=2) to choose. However, this might not always be the case with multiple readable storages.

Reply:
Updated this to remove the reverse lookup thing and to check instance type instead. This solution would not work for multiple readable storages though. But perhaps we don't need to design for that yet.